### PR TITLE
fixes CC-705

### PR DESCRIPTION
**Pull request by wwwtyro**

### CODE REVIEW

File: zzk/connection.go [hunk 1]
```diff
@@ -30,10 +31,8 @@ const (
 
 var (
 	ErrNotInitialized = errors.New("client not initialized")
-)
-
-var (
-	manager = make(map[string]*zclient)
+	manager           = make(map[string]*zclient)
+	managerLock       = &sync.Mutex{}
```
Comment (wplu) [inline zzk/connection.go]: use RWMutex

File: zzk/connection.go [hunk 2]
```diff
@@ -59,12 +58,16 @@ func GeneratePoolPath(poolID string) string {
...
 // InitializeLocalClient initializes the local zookeeper client
 func InitializeLocalClient(client *client.Client) {
+	managerLock.Lock()
+	defer managerLock.Unlock()
 	manager[local] = &zclient{client, make(map[string]*zconn)}
 }
 
 // GetLocalConnection acquires a connection from the local zookeeper client
 func GetLocalConnection(path string) (client.Connection, error) {
+	managerLock.Lock()
```
Comment (wplu) [inline zzk/connection.go]: managerLock.RLock()

File: zzk/connection.go [hunk 3]
```diff
@@ -59,12 +58,16 @@ func GeneratePoolPath(poolID string) string {
...
 	localclient, ok := manager[local]
+	managerLock.Unlock()
```
Comment (wplu) [inline zzk/connection.go]: managerLock.RUnlock()

File: zzk/connection.go [hunk 4]
```diff
@@ -73,12 +76,16 @@ func GetLocalConnection(path string) (client.Connection, error) {
...
 // InitializeRemoteClient initializes the remote zookeeper client
 func InitializeRemoteClient(client *client.Client) {
+	managerLock.Lock()
+	defer managerLock.Unlock()
 	manager[remote] = &zclient{client, make(map[string]*zconn)}
 }
 
 // GetRemoteConnection acquires a connection from the remote zookeeper client
 func GetRemoteConnection(path string) (client.Connection, error) {
+	managerLock.Lock()
```
Comment (wplu) [inline zzk/connection.go]: ditto

File: zzk/connection.go [hunk 5]
```diff
@@ -73,12 +76,16 @@ func GetLocalConnection(path string) (client.Connection, error) {
...
 	client, ok := manager[remote]
+	managerLock.Unlock()
```
Comment (wplu) [inline zzk/connection.go]: ditto

**Comment by wwwtyro** [comment]
test this

**Pull request merged**